### PR TITLE
chore(roll): roll to Playwright v1.45

### PR DESCRIPTION
**Pull request by mxschmitt**
To be discussed:

1. APIRequestContext.DisposeAsync(reason=...) is not ported yet - I also saw that we don't have BrowserContext.close(reason...) in .NET
2. We should evaluate if:
  ```
        Time = clone.Time;
        TimeDate = clone.TimeDate;
        TimeInt64 = clone.TimeInt64;
        TimeString = clone.TimeString;
  ```
  is expected or if we should call it `TimeLong` instead.

https://github.com/microsoft/playwright-dotnet/issues/2941

### CODE REVIEW

File: src/Playwright/Core/BrowserContext.cs [hunk 1]
```diff
@@ -130,6 +132,12 @@ public ITracing Tracing
         set => throw new NotSupportedException();
     }
 
+    public IClock Clock
+    {
+        get => _clock;
+        set => throw new NotSupportedException();
```
Comment (campersau) [inline src/Playwright/Core/BrowserContext.cs:138]: Is the setter needed?

File: src/Playwright/Core/Clock.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+internal class Clock(BrowserContext browserContext) : IClock
+{
+    public async Task InstallAsync(ClockInstallOptions options = null)
+    {
+        var timeIsGiven = options?.Time != null || options?.TimeInt64 != null || options?.TimeString != null || options?.TimeDate != null;
+        await browserContext.SendMessageToServerAsync("clockInstall", timeIsGiven ? ParseTime(options.Time ?? options.TimeString, options?.TimeDate, options?.TimeInt64) : null).ConfigureAwait(false);
+    }
+
+    private Dictionary<string, object> ParseTime(string timeString, DateTime? timeDate, long? timeInt64)
```
Comment (campersau) [inline src/Playwright/Core/Clock.cs:39]: Nit static.
```suggestion
    private static Dictionary<string, object> ParseTime(string timeString, DateTime? timeDate, long? timeInt64)
```

File: src/Playwright/API/Generated/Options/ClockInstallOptions.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        Time = clone.Time;
+        TimeDate = clone.TimeDate;
+        TimeInt64 = clone.TimeInt64;
+        TimeString = clone.TimeString;
+    }
+
+    /// <summary><para>Time to initialize with, current system time by default.</para></summary>
+    [JsonPropertyName("time")]
+    public string? Time { get; set; }
```
Comment (yury-s) [inline src/Playwright/API/Generated/Options/ClockInstallOptions.cs:51]: Why do we have both Time and TimeDate?

File: src/Playwright/Core/Clock.cs (hunk 2 above)
Comment (yury-s) [inline src/Playwright/Core/Clock.cs:39]: why not 3 different overloaded methods?

**Review by yury-s — APPROVED** [review]

File: src/Playwright/Core/BrowserContext.cs [hunk 4]
```diff
@@ -124,11 +126,9 @@ internal BrowserContext(ChannelOwner parent, string guid, BrowserContextInitiali
...
     public event EventHandler<IWorker> ServiceWorker;
 
-    public ITracing Tracing
-    {
-        get => _tracing;
-        set => throw new NotSupportedException();
-    }
+    public ITracing Tracing { get => _tracing; }
+
+    public IClock Clock { get => _clock; }
```
Comment (campersau) [inline src/Playwright/Core/BrowserContext.cs:131]: Nit: Could be even shorter like the other ones:

```suggestion
    public ITracing Tracing => _tracing;

    public IClock Clock => _clock;
```

File: src/Playwright/Core/Page.cs [hunk 5]
```diff
@@ -182,6 +182,11 @@ public IMouse Mouse
         get;
     }
 
+    public IClock Clock
+    {
+        get => Context.Clock;
+    }
```
Comment (kblok) [inline src/Playwright/Core/Page.cs:188]:
```suggestion
    public IClock Clock => Context.Clock;
```

**Pull request merged**